### ISSUE TITLE
IntelliJ / Cursive "forgets" the module source directory setting

### ISSUE THREAD

**Issue by karimarttila** [cursive-ide/cursive#2506]
IntelliJ Ultimate 2020.3.2 and Cursive 1.10.1-eap2-2020.3 on Ubuntu 20.04
I have directory structure:
- src
- - clj
- - cljc
- - cljs

I set the clj, cljc and cljs directories as "source" directories (Right mouse -> Mark directory as -> Sources Root) (or the same thing using File -> Project structure -> Project settings -> Modules ...)
I close IntelliJ.
I open IntelliJ again => the directories are not marked as "source" directories (blue) => I have to mark them as "source" directories again.

NOTE: It happens only occasionally.

**Comment by cursive-ide** [comment]
I assume that this is a project managed with either deps or leiningen? If so, then you have to add the directories as source roots in either your `deps.edn` or `project.clj`. Otherwise, next time Cursive syncs the project (e.g. on IntelliJ restart, or reopening the project) those settings will be overwritten again. Basically, on every sync they're updated to match the source of truth (the project file).

**Comment by karimarttila** [comment]
Damn. I just realized something. It must be as you said. In deps.edn:
`:backend {:extra-paths ["src/clj"]`
etc. 
And if those aliases are not checked in the IntelliJ/Cursive: Clojure Deps -> Aliases ...
... then in Cursive sync the directories are reset.
Damn. I just tested and it is exactly like that, how did I miss it?
- Remove alias from Clojure Deps -> Aliases
- deps.edn -> Refresh clojure deps projects
- The directory (e.g. clj) disappears to be as part of the source directories.

Thanks!
